plugins/uefi/fu-plugin-uefi.c

```diff
@@ -613,9 +612,63 @@ load_custom_esp (FuPlugin *plugin, GError **error)
 
 			return FALSE;
 		}
+#ifdef HAVE_FWUP_CUSTOM_ESP
+		fwup_set_esp_mountpoint (data->esp_path);
 		g_debug ("%s set to %s", key, data->esp_path);
-		fu_plugin_add_report_metadata (plugin, key,
-					       data->esp_path);
+#endif
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_uefi_delete_old_capsules (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	const gchar *pattern = "/boot/efi/EFI/*/fw/fwupdate-*.cap";
+	g_autoptr(GPtrArray) files = NULL;
+
+	/* delete any files matching the glob in the ESP */
+	files = fu_common_get_files_recursive (data->esp_path, error);
+	if (files == NULL)
+		return FALSE;
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch (pattern, fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
+				return FALSE;
+		}
+	}
+	return TRUE;
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
```

Comment:
It should be better to only run this new code if coldplug was successful in enumerating the esrt I would think rather than startup to keep fwupd startup quick on non esrt systems

Reply:
My eye starts to twitch when people talk about performance without numbers, but I do get what you mean. What about moving the check fwup_supported call to setup() and saving the `gint supported` in the plugin private data?

Reply:
That works too.